sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -313,131 +112,254 @@ object ScalaNativePluginInternal {
     },
     addCompilerPlugin(
       "org.scala-native" % "nscplugin" % nativeVersion cross CrossVersion.full),
-    nativeSharedLibrary := false,
-    nativeClang := None,
-    nativeClangPP := None,
+    nativeClang := {
```

Comment:
A bit of background: previously @MasseGuillaume suggested we make `nativeClang` a setting of type `Option[File]` instead of `File`. This seemed to be okey when most of the code was outside DSL world in helper functions. Now, when everything is a task, it's super annoying to deal with this option and it complicates code considerably. So instead, this PR goes with a simple solution: making `nativeClang` to be a task.